### ISSUE TITLE
BUG: Document links and the case information tab disappear to Practitioners associated to cases after the case is newly consolidated

### ISSUE THREAD

**Issue by ttlenard** [ustaxcourt/ef-cms#8568]
**Describe the Bug**
We received a helpdesk ticket from a Private Practitioner that reported that they had access to the case documents and the case information tab prior to their cases being consolidated.  This does not appear to be happening with private practitioners associated to cases that were previously consolidated.

I tested this with an IRS Practitioner as well. IRS Practitioners lose access to the documents, but still have access to the case information tab after cases are newly consolidated.

I also tested this with Petitioners, and this bug does not appear to have affected Petitioner access to documents or the case info tab. Yay!

**Business Impact/Reason for Severity**
Critical

**In which environment did you see this bug?**
Test, Prod

**Who were you logged in as?**
Private Practitioner, IRS Practitioner

**What were you doing when you discovered this bug? (Using the application, demoing, smoke tests, testing other functionality, etc.)**
This was reported to us by a private practitioner

**To Reproduce**
1. Be sure that you have access to a user that is associated with a few cases already
2. Be sure that these cases are in a state that they can be consolidated together (general docket - at issue ready for trial, same requested place of trial)
3. With this user, review the two cases and ensure that you have access to the case info tab, and can view the documents on the docket record.
4. Log in as a docket clerk
5. Navigate to one of the cases
6. Click on the case info tab
7. Type in the other case docket number to consolidated the cases
8. Check the box to consolidate the cases
9. Log in as the party
10. Click on one of the docket numbers for one of the cases that was just consolidated
11. Notice that you no longer have access to the documents on the docket record
12. Notice that the case information tab has disappeared
13. Note that you CAN still file a document on the case

Repeat the steps 1-8 above, but with an IRS Practitioner:
9. Log in as an IRS practitioner
10. Notice that you no longer have access to the documents on the docket record (with the exception of Orders)
11. Notice that the case information tab is still present
12. Note that you can still file a document on the case

**Expected Behavior**
Practitioner access should not be changed after newly consolidating cases

**Actual Behavior**
A clear and concise description of what actually happened.
Private Practitioner access to the case information tab and documents is removed after cases are newly consolidated.
IRS Practitioner loses access to case documents after cases are newly consolidated.  They can still see the case info tab.
This does not seem to have affected petitioner access.

**Screenshots**
This is a screen grab of a case that has a private practitioner associated with it, and it was newly consolidated.  Private Practitioner loses access to the case info tab as well as document links.
[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNENLQVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--f11524c3a085a9266b218cd1783d575f7e0c5f6a/image.png)

This is a screen grab of a case that has an IRS practitioner associated with it, and it was newly consolidated.  IRS Practitioner loses access to the case documents, but the case info tab is still there.
[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBN2FLQVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--1ecbeb2518f0877a4364f379f35e2a2061a02356/image.png)

This screen grab shows what a Petitioner sees. This bug does NOT seem to have affected Petitioner access.
[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBOFNLQVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--77be3f862422db08768db7e32dbe85b655a7b66e/image.png)

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Cause of Bug, If Known**

**Process for Logging a Bug:**
* Complete the above information
* Add a severity tag (Critical, High Severity, Medium Severity or Low Severity). See below for priority definition. 

**Severity Definition:**
* Critical Defect
Blocks entire system's or module’s functionality
No workarounds available
Testing cannot proceed further without bug being fixed.

* High-severity Defect
Affects key functionality of an application
There's a workaround, but not obvious or easy
App behaves in a way that is strongly different from the one stated in the requirements

* Medium-severity Defect
A minor function does not behave in a way stated in the requirements.
Workaround is available and easy

* Low-severity Defect
Mostly related to an application’s UI
Doesn't need a workaround, because it doesn't impact functionality

## Definition of Ready for Bugs(Created 10-4-21)
Definition used: A failure or flaw in the system which produces an incorrect or undesired result that deviates from the expected result or behavior. (Note: Expected results are use cases that have been documented in past user stories as acceptance criteria and test cases, and do not include strange behavior unrelated to use cases.)

The following criteria must be met in order for the development team to begin work on the bug.

The bug must:
 - Be focused on solving a user problem
 - Contain data for all fields in the bug template, so the team can pick it up and begin working immediately

Process: If the unexpected results are new use cases that have been identified, but not yet built, new acceptance criteria and test cases should be captured in a new user story and prioritized by the product owner. 

If the Court is not able to reproduce the bug, add the “Unable to reproduce” tag. This will provide visibility into the type of support that may be needed by the Court. In the event that the Court cannot reproduce the bug, the Court will work with Flexion to communicate what type of troubleshooting help may be needed.

## Definition of Done (Updated 4-14-21)
**Product Owner**
 - [ ]  Bug fix has been validated in the Court's test environment

**Engineering**
 - [ ] Automated test scripts have been written
 - [ ] Field level and page level validation errors (front-end and server-side) integrated and functioning
 - [ ] Verify that language for docket record for internal users and external users is identical
 - [ ] New screens have been added to pa11y scripts
 - [ ] All new functionality verified to work with keyboard and macOS voiceover https://www.apple.com/voiceover/info/guide/_1124.html 
 - [ ] READMEs, other appropriate docs and swagger/APIs fully updated
 - [ ] UI should be touch optimized and responsive for external only (functions on supported mobile devices and optimized for screen sizes as required)
 - [ ] Interactors should validate entities before calling persistence methods
 - [ ] Code refactored for clarity and to remove any known technical debt
 - [ ] Deployed to the Court's `test` environment if prod-like data is required. Otherwise, deployed to any `experimental` environment for review.

**Comment by ttlenard** [comment]
I found while testing that if a docket clerk removes the practitioner, and then re-adds them back on the case, the visibility to documents and the case information tab on the cases are restored.

**Comment by ttlenard** [comment]
Tom mentioned that there were two scenarios that caused the visibility issues.
1- Newly consolidated cases
2 - If a new case was added to an existing group, and the new case had a lower docket number, making it the new lead case.

Edit by Tom: In scenario 2 above, the practitioners already associated with the group would lose association across the group, while practitioners on the newly added case would not receive group association even if the new case was not the lowest numbered

**Comment by ttlenard** [comment]
@TomElliottFlexion Testing looks good. Thank you!

**Comment by ttlenard** [comment]
Note: This bugfix will not fix the visibility on cases that were consolidated between 10/21 and now. We have recommended to the Court to not consolidate any additional cases until this fix is in place in production.  After reviewing the "Motion to consolidate GRANTED" Orders that occurred between 10/21 and today, we found 2 groups were consolidated.  We have already manually fixed one group by having the court remove the practitioners and add them back in. This fixed the visibility issue.  We will need to manually fix the other group so that visibility is restored for the practitioners on those cases.  I will be reaching out to the Court with this guidance today.
